### PR TITLE
[kesch] Kesch+pgi

### PR DESCRIPTION
**Pull request by jgphpc**
for https://github.com/eth-cscs/production/pull/809

### CODE REVIEW

File: easybuild/easyconfigs/p/pgi/pgi-17.10-gcc-5.4.0-2.26.eb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+binutilsver = '2.26'
+versionsuffix = '-GCC-%s-%s' % (gccver, binutilsver)
+
+homepage = 'http://www.pgroup.com/'
+description = "C, C++ and Fortran compilers from The Portland Group - PGI"
+
+toolchain = {'name': 'dummy', 'version': ''}
+
+dependencies = [
+    ('gcc/5.4.0-2.26', EXTERNAL_MODULE),
```
Comment (kraushm) [inline easybuild/easyconfigs/p/pgi/pgi-17.10-gcc-5.4.0-2.26.eb]: Would it make it more easy to maintain this, if the variables defined above were also used for the dependencies?

Comment (gppezzi) [reply easybuild/easyconfigs/p/pgi/pgi-17.10-gcc-5.4.0-2.26.eb]: this module is provided by EB, why are you using as external module?

**Comment by gppezzi** [comment]
I suggest to include this file in the same PR as netcdf. 

If this is only used by netcdf (as dependency), you don't need to add it to the production list, only adding netcdf should be enough.

File: easybuild/easyconfigs/p/pgi/pgi-17.10-gcc-5.4.0-2.26.eb (hunk 1 above)
Comment (jgphpc) [reply easybuild/easyconfigs/p/pgi/pgi-17.10-gcc-5.4.0-2.26.eb]: this is what i get if i don't put it as external:
```
[kesch] ERROR: Irresolvable dependencies encountered: gcc/5.4.0-2.26
```

`/apps/escha/UES/PrgEnv-gnu-17.02/modulefiles/gcc/5.4.0-2.26`

Comment (gppezzi) [reply easybuild/easyconfigs/p/pgi/pgi-17.10-gcc-5.4.0-2.26.eb]: GCC is capital in EB.

MCH asked to have a lowercase naming scheme so the module names are misleading, we need always to look at the eb files.

File: easybuild/easyconfigs/p/pgi/pgi-17.10-gcc-5.4.0-2.26.eb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+binutilsver = '2.26'
+versionsuffix = '-GCC-%s-%s' % (gccver, binutilsver)
+
+homepage = 'http://www.pgroup.com/'
+description = "C, C++ and Fortran compilers from The Portland Group - PGI"
+
+toolchain = {'name': 'dummy', 'version': ''}
+
+dependencies = [
+    ('GCC', '5.4.0-2.26'),
```
Comment (kraushm) [inline easybuild/easyconfigs/p/pgi/pgi-17.10-gcc-5.4.0-2.26.eb]: I would still vote for using the variables `gccver` and `binutilsver` also for the gcc dependency.

File: easybuild/easyconfigs/p/pgi/pgi-17.10-gcc-5.4.0-2.26.eb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    ('pgi/17.10', EXTERNAL_MODULE),
```
Comment (kraushm) [inline easybuild/easyconfigs/p/pgi/pgi-17.10-gcc-5.4.0-2.26.eb]: Same argument, only for the variable `version`.

File: easybuild/easyconfigs/p/pgi/pgi-17.10-gcc-5.4.0-2.26.eb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+easyblock = 'Bundle'
+
+name = 'PGI'
```
Comment (gppezzi) [inline easybuild/easyconfigs/p/pgi/pgi-17.10-gcc-5.4.0-2.26.eb]: The eb file name here does not match the case of the package name , so you might have problems when using this eb as dependency (one more reason for having this file in the same pr as netcdf).

**Pull request merged**